File: DTOs/Sources/AuthServiceDTOs/UserEmailVerificationDTO.swift
import Foundation

public struct UserEmailVerificationDTO: Codable {
    public var uid: UUID?
    public var name: String?
    public var isActive: Bool?
    public var emailStatus: VerificationStatus?
    public var createdAt: Date?

    public init(uid: UUID? = nil, name: String? = nil, isActive: Bool? = nil, emailStatus: VerificationStatus? = nil, createdAt: Date? = nil) {
        self.uid = uid
        self.name = name
        self.isActive = isActive
        self.emailStatus = emailStatus
        self.createdAt = createdAt
    }

}

public enum VerificationStatus: String, Codable, CaseIterable {
    case failed
    case pending
    case verified
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/AuthServiceDTOs/UserEmailVerificationDTO.kt
package net.ipv64.kivop.dtos.AuthServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class UserEmailVerificationDTO (
    var uid : UUID?,
    var name : String?,
    var isActive : Boolean?,
    var emailStatus : VerificationStatus?,
    var createdAt : LocalDateTime?,
)
enum class VerificationStatus {
    failed,
    pending,
    verified,
}
